File: Testing/TestProcessorToolbox.cc
/*
 * TestProcessorToolbox.cc
 *
 *  Created on: Nov 25, 2021
 *      Author: N.S. Oblath
 */

#include "TestProc.hh"

#include "ProcessorToolbox.hh"

#include "logger.hh"
#include "param_codec.hh"

#include "catch.hpp"

LOGGER( testlog, "TestProcessorToolbox" );


TEST_CASE( "processor_toolbox" )
{
    using namespace Nymph;

    ProcessorToolbox toolbox;

    SECTION( "Configuration" )
    {
        REQUIRE_FALSE( toolbox.GetProcessor( "testproc-1" ) );

        std::string config_str(
            "processors:"
            "- type: test-proc"
            "  name: testproc-1"
        );

        scarab::param_translator translator;
        auto config = translator.read_string( config_str );

        toolbox.Configure( config->as_node() );

        REQUIRE( toolbox.GetProcessor( "testproc-1" ) );


    }


}

File: Testing/Python/testprocessortoolbox.py

"""
  testprocessortoolbox.py
 
   Created on: Jun 14, 2022
       Author: N.S. Oblath
"""
import unittest

import _nymph, _nymph_testing


class TestDataMethods(unittest.TestCase):
    tb = _nymph.processor.ProcessorToolbox()

    procName1 = 'testproc-1'  # not a registered processor
    procName2 = 'testproc-2'  # not a registered processor
    procName3 = 'test-proc'   # a registered processor

    '''Testing Processor handling'''
    def test_add_remove_processors(self):
        self.assertFalse( self.tb.has_processor( self.procName1 ) )
        self.assertFalse( self.tb.has_processor( self.procName2 ) )
        self.assertFalse( self.tb.has_processor( self.procName3 ) )
        self.assertFalse( self.tb.could_build( self.procName1 ) )
        self.assertFalse( self.tb.could_build( self.procName2 ) )
        self.assertTrue( self.tb.could_build( self.procName3 ) )
    

if __name__ == '__main__':
    unittest.main()
